internal/implementations/gateway/implementation_suit_test.go

```diff
@@ -0,0 +1,13 @@
+package implementation_test
+
+import (
+	"testing"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+func TestState(t *testing.T) {
```

Comment:
@kate-osborn +1 for consistency

I don't remember the original intention of naming them all implementation, rather than "service" or "gatewayconfig". 

Pinging @f5yacobucci